### PR TITLE
Optimize `scripts/prohibit-properties-unused-key`

### PR DESCRIPTION
**Pull request by PawelLipski**
Before:
```
$ time scripts/prohibit-properties-unused-keys

real	0m1.319s
user	0m0.832s
sys	0m1.896s

$ time .git/hooks/pre-commit
.........................
real	0m2.343s
user	0m1.685s
sys	0m2.939s

```

After:
```
$ time scripts/prohibit-properties-unused-keys

real	0m0.061s
user	0m0.139s
sys	0m0.023s

$ time .git/hooks/pre-commit
......................
real	0m1.044s
user	0m0.932s
sys	0m0.966s
```

### CODE REVIEW

File: scripts/prohibit-properties-unused-keys [hunk 1]
```diff
@@ -8,14 +8,21 @@ source "$self_dir"/utils.sh
 bundle=frontend/base/src/main/resources/GitMacheteBundle.properties
 [[ -f $bundle ]] || die "Bundle file ($bundle) has been (re)moved"
 
-exit_code=0
-for key in $(
-  grep -Po '.*(?==)' -- $bundle \
-  | grep -Evx 'action\.GitMachete\.[A-Za-z]+\.(description|GitMachete(ContextMenu|Toolbar).text|text)'
-); do
-  if ! git grep -qF "$key" -- '*.java' '*.kt' ":!$bundle"; then
```
Comment (PawelLipski) [inline scripts/prohibit-properties-unused-keys:16]: `git grep` executed for each property key separately... Not Good ™️

File: scripts/prohibit-properties-unused-keys [hunk 2]
```diff
@@ -8,14 +8,21 @@ source "$self_dir"/utils.sh
...
+    | sort
+)
+
+all_found_keys=$(
+  git grep -h --only-matching --fixed-strings "$all_expected_keys" -- '*.java' '*.kt' ":!$bundle" \
+    | sort | uniq
+)
+
+# Print the lines present in the first output but missing from the second
+output=$(comm -23 <(echo "$all_expected_keys") <(echo "$all_found_keys"))
```
Comment (PawelLipski) [inline scripts/prohibit-properties-unused-keys:23]: https://virtuslab.github.io/tips/#shell/output-comparing

File: scripts/prohibit-properties-unused-keys (hunk 1 above)
Comment (PawelLipski) [reply scripts/prohibit-properties-unused-keys:16]: https://virtuslab.github.io/tips/#regex/lookbehind-lookahead, IMO the only advanced regex feature (assuming that capturing groups aren't considered advanced) you'll ever need

File: scripts/prohibit-properties-unused-keys [hunk 3]
```diff
@@ -8,14 +8,21 @@ source "$self_dir"/utils.sh
...
-  | grep -Evx 'action\.GitMachete\.[A-Za-z]+\.(description|GitMachete(ContextMenu|Toolbar).text|text)'
-); do
-  if ! git grep -qF "$key" -- '*.java' '*.kt' ":!$bundle"; then
-    echo "Property key $key defined in $bundle does not seem to be used anywhere"
-    exit_code=1
-  fi
-done
-exit $exit_code
+all_expected_keys=$(
+  grep --only-matching --perl-regexp '.*(?==)' -- $bundle \
```
Comment (PawelLipski) [inline scripts/prohibit-properties-unused-keys:12]: https://virtuslab.github.io/tips/#regex/lookbehind-lookahead, IMO the only advanced regex feature (assuming that capturing groups aren't considered advanced) you'll ever need

File: scripts/prohibit-properties-unused-keys [hunk 4]
```diff
@@ -8,14 +8,22 @@ source "$self_dir"/utils.sh
...
+  git grep -h --only-matching --fixed-strings "$all_expected_keys" -- '*.java' '*.kt' ":!$bundle" \
+    | sort | uniq
+)
+
+# Print the lines present in the first output but missing from the second
+output=$(comm -23 <(echo "$all_expected_keys") <(echo "$all_found_keys"))
+
+if [[ $output ]]; then
+  echo "$output"
+  die "The above property keys defined in $bundle does not seem to be used anywhere"
```
Comment (MaciejG604) [inline scripts/prohibit-properties-unused-keys:28]:
```suggestion
  die "The above property keys defined in $bundle do not seem to be used anywhere"
```

**Review by mkondratek — APPROVED** [review]

**Pull request merged**